### PR TITLE
Added support of filter for register environment

### PR DESCRIPTION
**Pull request by chshrikh**

### CODE REVIEW

**Comment by KrishnaAdityaB** [comment]
Signed-off

File: Tasks/WindowsMachineFileCopy/WindowsMachineFileCopy.ps1 [hunk 1]
```diff
@@ -186,25 +149,19 @@ else
...
-        $resources = Get-EnvironmentResources -EnvironmentName $fetchedEnvironmentName -TagFilter $wellFormedTagsList -TaskContext $distributedTaskContext
-        Write-Verbose "Completed Get-EnvironmentResources cmdlet call for environment name: $fetchedEnvironmentName with tag filter" -Verbose
-    }
-    else
+    if ($resources.Count -eq 0)
     {
-        Write-Verbose "Starting Get-EnvironmentResources cmdlet call on environment name: $fetchedEnvironmentName with machine filter: $machineFilter" -Verbose
-        $resources = Get-EnvironmentResources -EnvironmentName $fetchedEnvironmentName -ResourceFilter $machineFilter -TaskContext $distributedTaskContext
-        Write-Verbose "Completed Get-EnvironmentResources cmdlet call for environment name: $fetchedEnvironmentName with machine filter" -Verbose
+         throw (Get-LocalizedString -Key "No machine exists under environment: '{0}' for deployment" -ArgumentList $environmentName)
```
Comment (kmkumaran) [inline Tasks/WindowsMachineFileCopy/WindowsMachineFileCopy.ps1]: User facing messages should not have 'environment' word. a) environment has a different meaning in RM. b) I am not sure if the notion of environment is meaningful when a flat list is given (will this message ever come for flatlist or variable?)

File: Tasks/WindowsMachineFileCopy/WindowsMachineFileCopy.ps1 [hunk 2]
```diff
@@ -186,25 +149,19 @@ else
 
     $connection = Get-VssConnection -TaskContext $distributedTaskContext
 
-    Write-Verbose "Starting Register-Environment cmdlet call for environment : $environmentName" -Verbose
-    $environment = Register-Environment -EnvironmentName $environmentName -EnvironmentSpecification $environmentName -UserName $adminUserName -Password $adminPassword -Connection $connection -TaskContext $distributedTaskContext
+    Write-Verbose "Starting Register-Environment cmdlet call for environment : $environmentName with filter $machineFilter" -Verbose
+    $environment = Register-Environment -EnvironmentName $environmentName -EnvironmentSpecification $environmentName -UserName $adminUserName -Password $adminPassword -Connection $connection -TaskContext $distributedTaskContext -ResourceFilter $machineFilter
     Write-Verbose "Completed Register-Environment cmdlet call for environment : $environmentName" -Verbose
 
     $fetchedEnvironmentName = $environment.Name
```
Comment (kmkumaran) [inline Tasks/WindowsMachineFileCopy/WindowsMachineFileCopy.ps1]: It's not a blocker for M91. But can we rename this variable that is more intuitive. It is not obvious (that might be cloning) as to why you have to get the name when it was already supplied as input.

File: Tasks/WindowsMachineFileCopy/WindowsMachineFileCopy.ps1 (hunk 1 above)
Comment (chshrikh) [inline Tasks/WindowsMachineFileCopy/WindowsMachineFileCopy.ps1]: sure , I will take care if this comment with next checkin

File: Tasks/WindowsMachineFileCopy/WindowsMachineFileCopy.ps1 (hunk 2 above)
Comment (chshrikh) [inline Tasks/WindowsMachineFileCopy/WindowsMachineFileCopy.ps1]: sure we will consider this with next checkin

**Pull request merged**